crates/proof-of-sql/src/base/math/decimal.rs

```diff
@@ -1,10 +1,51 @@
 //! Module for parsing an `IntermediateDecimal` into a `Decimal75`.
 use crate::{
-    base::scalar::Scalar,
-    sql::parse::{ConversionError, ConversionResult},
+    base::{
+        math::decimal::DecimalError::{
+            IntermediateDecimalConversionError, InvalidPrecision, RoundingError,
+        },
+        scalar::Scalar,
+    },
+    sql::parse::{
+        ConversionError::{self, DecimalConversionError},
+        ConversionResult,
+    },
 };
-use proof_of_sql_parser::intermediate_decimal::IntermediateDecimal;
+use proof_of_sql_parser::intermediate_decimal::{IntermediateDecimal, IntermediateDecimalError};
 use serde::{Deserialize, Deserializer, Serialize};
+use thiserror::Error;
+
+/// Errors related to decimal operations.
+#[derive(Error, Debug, Eq, PartialEq)]
+pub enum DecimalError {
+    #[error("Invalid decimal format or value: {0}")]
+    /// Error when a decimal format or value is incorrect,
+    /// the string isn't even a decimal e.g. "notastring",
+    /// "-21.233.122" etc aka InvalidDecimal
+    InvalidDecimal(String),
+
+    #[error("Decimal precision is not valid: {0}")]
+    /// Decimal precision exceeds the allowed limit,
+    /// e.g. precision above 75/76/whatever set by Scalar
+    /// or non-positive aka InvalidPrecision
+    InvalidPrecision(String),
+
+    #[error("Unsupported operation: cannot round decimal: {0}")]
+    /// This error occurs when attempting to scale a
+    /// decimal in such a way that a loss of precision occurs.
+    RoundingError(String),
+
+    /// Errors that may occur when parsing an intermediate decimal
+    /// into a posql decimal
+    #[error("Intermediate decimal conversion error: {0}")]
+    IntermediateDecimalConversionError(IntermediateDecimalError),
+}
+
+impl From<IntermediateDecimalError> for ConversionError {
+    fn from(err: IntermediateDecimalError) -> ConversionError {
+        DecimalConversionError(IntermediateDecimalConversionError(err))
+    }
+}
```

Comment:
Please use #[from]

Reply:
we cannot do that without putting the variant back into ```ConversionError```